src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -440,15 +440,21 @@ private static string ComposeMessageFromInnerExceptions(Exception exception)
                     package.HasCompileTimePlaceholder &&
                     !TargetFramework.IsPackageBased)
                 {
+                    // requiresFrameworkAssemblies is true whenever we find a CompileTimePlaceholder in a non-package based framework, even if
+                    // the reference is unresolved. This ensures the best error experience when someone is building on a machine without
+                    // the target framework installed.
+                    requiresFrameworkAssemblies = true;
```

Comment:
Should this be called "mayRequireFrameworkAssemblies"?  This will occur when folks use `exclude="compile"` which we commonly do in CoreFx packages (as does Roslyn which caused this issue).  Probably not a serious issue now since these are all likely to require a targeting pack, but this might need to be revisited when we look at replacing targeting packs with packages.